### PR TITLE
relnote(108): support for 'effective-directive' and 'status-code' in violation reports

### PR DESCRIPTION
**Pull request by bsmth**
CSP violation reports sent by Firefox include `effective-directive` and `status-code` properties in the payload.

__Additions:__
* Notes that the `violated-directive` is a historic name for the `effective-directive` field and contains the same value, see comment in https://w3c.github.io/webappsec-csp/#violation-events syntax
* `report-uri` is deprecated in favor of `report-to` directive, see https://w3c.github.io/webappsec-csp/#changes-from-level-2

__Related issues:__
- [x] https://github.com/mdn/content/issues/22907

### CODE REVIEW

File: files/en-us/web/http/csp/index.md [hunk 1]
```diff
@@ -184,15 +186,15 @@ The report JSON object contains the following data:
 - `status-code`
   - : The HTTP status code of the resource on which the global object was instantiated.
 - `violated-directive`
```
Comment (hamishwillee) [inline files/en-us/web/http/csp/index.md:188]: The current name should be referenced, not the historic one IMO. You will need to check that "violated-directive" anchor is not linked from elsewhere.
```suggestion
- `effective-directive`
```

File: files/en-us/web/http/headers/content-security-policy-report-only/index.md [hunk 2]
```diff
@@ -83,14 +83,14 @@ The report JSON object contains the following data:
 - `status-code`
   - : The HTTP status code of the resource on which the global object was instantiated.
 - `violated-directive`
```
Comment (hamishwillee) [inline files/en-us/web/http/headers/content-security-policy-report-only/index.md:85]: As above

```suggestion
- `effective-directive`
```

**Review by hamishwillee — COMMENTED** [review]
Other than comment looks good to me. But I'm away and unavailable for discussion from today, so you probably need another reviewer

File: files/en-us/web/http/csp/index.md (hunk 1 above)
Comment (bsmth) [reply files/en-us/web/http/csp/index.md:188]: Good point, although both are included in a JSON payload with the same contents:

```json
{
   "csp-report":{
      "blocked-uri":"https://apis.google.com/js/platform.js",
      "disposition":"enforce",
      "document-uri":"https://localhost:5000/",
      "effective-directive":"script-src-elem",
      "original-policy":"default-src 'none'; report-uri https://localhost:5000/listener",
      "referrer":"",
      "status-code":200,
      "violated-directive":"script-src-elem"
   }
}
```

Comment (hamishwillee) [reply files/en-us/web/http/csp/index.md:188]: Yeah, but we document the one we want people to use. The alternative is to document both and mark "violated-directive" as deprecated. Your call.

Comment (bsmth) [reply files/en-us/web/http/csp/index.md:188]: I'm leaning towards leaving it in as it's hard to disambiguate the two during testing; all browsers will send this key in the payload and this took a lot of digging for me to find out if there was any effective difference.

File: files/en-us/web/http/headers/content-security-policy-report-only/index.md [hunk 3]
```diff
@@ -137,4 +139,5 @@ As you can see, the report includes the full path to the violating resource in `
 ## See also
 
 - {{HTTPHeader("Content-Security-Policy")}}
-- CSP {{CSP("report-uri")}} directive
+- CSP {{CSP("report-to")}} directive
+- CSP {{CSP("report-to")}} directive {{deprecated_inline}}
```
Comment (bsmth) [inline files/en-us/web/http/headers/content-security-policy-report-only/index.md:143]: Typo 

```suggestion
- CSP {{CSP("report-uri")}} directive {{deprecated_inline}}
```

File: files/en-us/web/http/csp/index.md (hunk 1 above)
Comment (hamishwillee) [reply files/en-us/web/http/csp/index.md:188]: @bsmth Ah, so I missed that the `effective-directive` was documented above - so you're right to leave this.

But what I will do here is add deprecated inline.

```suggestion
- `violated-directive` {{deprecated_inline}}
```

File: files/en-us/web/http/csp/index.md [hunk 4]
```diff
@@ -183,16 +185,16 @@ The report JSON object contains the following data:
     Only applicable to `script-src*` and `style-src*` violations, when they contain the `'report-sample'`
 - `status-code`
   - : The HTTP status code of the resource on which the global object was instantiated.
-- `violated-directive`
-  - : The name of the policy section that was violated.
+- `violated-directive` {{deprecated_inline}}
```
[the remark this answers is not in the document]
Comment (bsmth) [reply files/en-us/web/http/csp/index.md:188]: Great, thank you!

File: files/en-us/web/http/csp/index.md (hunk 1 above)
Comment (bsmth) [reply files/en-us/web/http/csp/index.md]: >But what I will do here is add deprecated inline.

Good call 👍🏻

**Review by hamishwillee — APPROVED** [review]
Great. Thank you.

**Pull request merged**